### ISSUE TITLE
platform: handle crash-looping clusters

### ISSUE THREAD

**Issue by sploiselle** [MaterializeInc/materialize#12300]
### Feature request

@philip-stoev found this in #12132, details [here](https://github.com/MaterializeInc/materialize/pull/12132#issuecomment-1115277083)

This:

```
CREATE TABLE t1 (f1 TEXT);
CREATE MATERIALIZED VIEW v1 AS SELECT mz_internal.mz_panic(f1) FROM t1;
CREATE CLUSTER REPLICA r2 FOR default SIZE '2-2';
INSERT INTO t1 VALUES ('a');

SELECT mz_internal.mz_sleep(10);
DROP VIEW v1;
```

causes a perpetual crash loop for a compute instance/cluster--i.e. it crashes and the orchestrator service continually reboots it, only to crash again.

At the moment, there are no measures (AFAIK) to mitigate this kind of behavior, so the cluster's replicas will all crash loop until the services are stopped somehow. We might want to e.g. create some kind of restart allowance that once exceeded causes the service to be removed entirely.

**Comment by philip-stoev** [comment]
In my opinion the problem here is that the `DROP VIEW` does not stop the crash-looping like it does on a single-node cluster.

Note that this particular situation requires the use of `mz_panic()` to reproduce. Just kill-ing random `computed`s seems to be handled correctly -- they are all restarted properly and no crash-looping occurs.

**Comment by philip-stoev** [comment]
This is no longer reproducible on main with materialized views, regardless of the cluster size. Issuing `DROP VIEW` successfully clears those situations.

However, the combination of `SELECT mz_internal.mz_panic(f1) FROM t1;` followed by `CTRL+Z` still results in a crash loop.

**Comment by benesch** [comment]
> However, the combination of SELECT mz_internal.mz_panic(f1) FROM t1; followed by CTRL+Z still results in a crash loop.

This is both fixed and tracked in #12091! I'm going to call this one closed, and we can use #12091 to track getting a test for this scenario.
